storagemarket/provider_loop.go

```diff
@@ -442,6 +446,23 @@ func (p *Provider) run() {
 			}
 			publishedDeal.done <- struct{}{}
 
+		case retryDealReq := <-p.retryDealChan:
+			// Get deal by uuid from the database
+			deal, err := p.dealsDB.ByID(p.ctx, retryDealReq.dealUuid)
+			if err != nil {
+				retryDealReq.done <- err
+				return
+			}
+
+			// start executing the deal
+			if p.startDealThread(deal) {
+				// if the deal wasn't already running, log a message saying
+				// that it was restarted
+				p.dealLogger.Infow(deal.DealUuid, "user initiated deal retry", "checkpoint", deal.Checkpoint)
+			}
```

Comment:
Maybe log an else statement that `deal is already running`? I think the absence of a log line will make it harder to figure out what's going on.